manifests/components/nginx-ingress.jsonnet

```diff
@@ -141,11 +141,23 @@ local NGNIX_INGRESS_IMAGE = (import "images.json")["nginx-ingress-controller"];
 
   hpa: kube.HorizontalPodAutoscaler($.p + "nginx-ingress-controller") + $.metadata {
     target: $.controller,
-    spec+: {maxReplicas: 10},
+    spec+: {
+      // Put a cap on growth due to (eg) DoS attacks.
+      // Large sites will want to increase this to cover legitimate demand.
+      maxReplicas: 10,
+    },
+  },
+
+  pdb: kube.PodDisruptionBudget($.p + "nginx-ingress-controller") + $.metadata {
+    target_pod: $.controller.spec.template,
+    spec+: {minAvailable: $.controller.spec.replicas - 1},
```

Comment:
This only allows one instance to be down at a time, right? Does NGINX support lame-duck mode?

Reply:
nginx-ingress controller supports lame-duck, and we configure the nginx pod with `terminationGracePeriodSeconds=60`.
Since I just checked, the controller is running as pid 1, so it will receive the SIGTERM, and it [catches that](https://github.com/kubernetes/ingress-nginx/blob/17e788b8e11536e037e52cfd56e5849e7bb5405f/cmd/nginx/main.go#L163) and turns it into a clean nginx shutdown.  It's not immediately clear to me [from the code](https://github.com/kubernetes/ingress-nginx/blob/17e788b8e11536e037e52cfd56e5849e7bb5405f/internal/ingress/controller/checker.go#L38) whether it fails healthz checks during that time or not.  (Actual observations welcome)